### PR TITLE
added nuget restore task in rake

### PR DESCRIPTION
**Pull request by jchannon**

### CODE REVIEW

File: rakefile.rb [hunk 1]
```diff
@@ -31,6 +31,18 @@
 CLEAN.include(FileList["src/**/#{CONFIGURATION}"])
 CLEAN.include(FileList["test/**/#{CONFIGURATION}"])
 
+desc "Restore NuGet packages"
+exec :nuget_restore do |cmd|
+   cmd.command = "tools/nuget/NuGet.exe"
```
Comment (thecodejunkie) [inline rakefile.rb]: Nope, that does not exist. It has to be `dependencies/Nancy/tools/nuget/NuGet.exe`

File: rakefile.rb [hunk 2]
```diff
@@ -31,6 +31,18 @@
...
+desc "Restore NuGet packages"
+exec :nuget_restore do |cmd|
+   cmd.command = "tools/nuget/NuGet.exe"
+   cmd.parameters = ["restore #{SOLUTION_FILE}"]
+end
+
+desc "Restore NuGet packages on Mono"
+exec :nuget_restore_mono do |cmd|
+  cmd.command = "mono"
+  cmd.parameters = ["tools/nuget/NuGet.exe restore #{SOLUTION_FILE}"]
```
Comment (thecodejunkie) [inline rakefile.rb]: Same here

File: rakefile.rb [hunk 3]
```diff
@@ -31,6 +31,18 @@
...
 CLEAN.include(FileList["test/**/#{CONFIGURATION}"])
 
+desc "Restore NuGet packages"
+exec :nuget_restore do |cmd|
+   cmd.command = "tools/nuget/NuGet.exe"
+   cmd.parameters = ["restore #{SOLUTION_FILE}"]
+end
+
+desc "Restore NuGet packages on Mono"
+exec :nuget_restore_mono do |cmd|
```
Comment (khellang) [inline rakefile.rb]: @thecodejunkie You didn't add this mono task in the other repos. I assume we don't use them anyway?

Comment (jchannon) [inline rakefile.rb]: Only by luck. See Slack :)

**Pull request merged**